ansible/roles/operator-pipeline/templates/openshift/tasks/acquire-lock.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: acquire-lease
+spec:
+  params:
+  - name: lease-name
+    type: string
+  - name: timeout
+    type: string
+    default: 90m
+  stepTemplate:
+    env:
+      - name: TOTAL_DURATION_IN_SECONDS
+        value: ${0}
```

Comment:
What's the purpose of this? I don't think it's needed.

Reply:
The TOTAL_DURATION_IN_SECONDS variable is needed to be an env variable because it wasn't straight forward to use the return value from one defined function to another defined function. After declaring the variable as an env variable, that issue got resolved.

Reply:
You are not declaring TOTAL_DURATION_IN_SECONDS as an env var here: you are telling tekton to run the script with TOTAL_DURATION_IN_SECONDS set to something undefined (either the literal string `${0}` or it will get resolved to the name of the shell being executed), so I don't think it makes a lot of sense as it is.
Since `calculate_duration_in_seconds()` is the first function you call, it will set the value of `TOTAL_DURATION_IN_SECONDS` for the rest of the script.
What you are missing is an `export TOTAL_DURATION_IN_SECONDS` at the end of `calculate_duration_in_seconds()` because you are executing `acquire_lease()` in a subshell.